src/Workspaces/Core/Portable/Storage/AbstractPersistentStorageService.cs

```diff
@@ -61,6 +61,11 @@ public IChecksummedPersistentStorage GetStorage(Solution solution, bool checkBra
                 return NoOpPersistentStorage.Instance;
             }
 
+            return GetStorageWorker(solution);
+        }
+
+        internal IChecksummedPersistentStorage GetStorageWorker(Solution solution)
```

Comment:
❔ Is there a reason this method is needed? I thought setting the solution size threshold and database location would be enough to avoid the use of `NoOpPersistentStorage`.

Reply:
Unfortunately no.  There is also an additional check that the workspace has a valid solution-file.  For an adhoc workspace this check fails and we don't create the db.  I don't want to change anything in real code about htat just now.